src/Selenium2Library/keywords/_element.py

```diff
@@ -602,41 +602,54 @@ def page_should_not_contain_image(self, locator, message='', loglevel='INFO'):
     def get_matching_xpath_count(self, xpath):
         """Returns number of elements matching `xpath`
 
-        One should not use the xpath= prefix for 'xpath'. XPath is assumed.
+        As of release 1.8 `Get Matching Xpath Count` allows the
+        argument `xpath` to start with the xpath= locator strategy.
 
-        Correct:
+        Example:
         | count = | Get Matching Xpath Count | //div[@id='sales-pop']
-        Incorrect:
         | count = | Get Matching Xpath Count | xpath=//div[@id='sales-pop']
 
         If you wish to assert the number of matching elements, use
         `Xpath Should Match X Times`.
         """
-        count = len(self._element_find("xpath=" + xpath, False, False))
-        return str(count)
+        if xpath.startswith('//'):
+            xpath = "xpath=" + xpath
+        
+        if xpath.startswith('xpath='):
+            count = len(self._element_find(xpath, False, False))
+            return str(count)
+        else:
+            raise ValueError("Element locator must be xpath and start with xpath= or //")
 
-    def xpath_should_match_x_times(self, xpath, expected_xpath_count, message='', loglevel='INFO'):
-        """Verifies that the page contains the given number of elements located by the given `xpath`.
+    def xpath_should_match_x_times(self, locator, expected_xpath_count, message='', loglevel='INFO'):
```

Comment:
If you want to avoid an issue with the named argument processing of RF, I would suggest calling the argument xpath_, rather than locator.
Adding a trailing underscore is the PEP 8 suggestion for when there is a conflict with Python Keywords.
BuiltIn's Sleep and Get Time keywords use a trailing namespace on the time argument.